### PR TITLE
Apply migrations in development containers

### LINKED ISSUE #194: Handle automatic migrations in development

### PR DESCRIPTION
**Pull request by kevincianfarini**
Until now, we've relied on manually dumping the schema of the production
database to synchronize our local development environments.

This commit will apply migrations via `flyway` as part of
`docker-compose up`.

Closes #194.

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -1,38 +1,55 @@
...
+      - POSTGRES_PASSWORD=${POSTGRES_PASSWORD}
+      - POSTGRES_USER=${POSTGRES_USER}
+      - POSTGRES_DB=${POSTGRES_DB} 
+      - POSTGRES_HOSTNAME=${POSTGRES_HOSTNAME}
+      - POSTGRES_PORT=${POSTGRES_PORT}
+    restart: on-failure:5
+    volumes: 
+      - ./backend:/flyway/backend:ro
+    depends_on: 
+      - database
```
Comment (kevincianfarini) [inline docker-compose.yml:55]: This is the new stuff. The above changes are simply an indentation change, as i realized it was improperly indented.

File: flyway/migrate.sh [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+# Allocate a new temporary directory for the migration files. 
+tmp_dir=$(mktemp -d -t migrations-XXXXXX)
+
+# Move the SQLDelight migrations to the temporary directory. 
+cp backend/src/main/sqldelight/migrations/*.sqm $tmp_dir 
+
+# Rename the .sqm files in tmp_dir to .sql files. This assumes that are valid SQL. 
+find $tmp_dir -name "*.sqm" -exec sh -c 'mv "$1" "${1%.sqm}.sql"' _ {} \;
+
+ls -la $tmp_dir
```
Comment (kevincianfarini) [inline flyway/migrate.sh:12]: Remove.

File: flyway/migrate.sh [hunk 3]
```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+
+# Allocate a new temporary directory for the migration files. 
+tmp_dir=$(mktemp -d -t migrations-XXXXXX)
+
+# Move the SQLDelight migrations to the temporary directory. 
+cp backend/src/main/sqldelight/migrations/*.sqm $tmp_dir 
+
+# Rename the .sqm files in tmp_dir to .sql files. This assumes that are valid SQL.
```
Comment (kevincianfarini) [inline flyway/migrate.sh:9]: typo.

File: flyway/migrate.sh [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+find $tmp_dir -name "*.sqm" -exec sh -c 'mv "$1" "${1%.sqm}.sql"' _ {} \;
+
+ls -la $tmp_dir
+
+# Build the jdbc URI.
+POSTGRES_URL="jdbc:postgresql://${POSTGRES_HOSTNAME}:${POSTGRES_PORT}/${POSTGRES_DB}"
+
+# Initiate the migration. 
+flyway -user=${POSTGRES_USER} \
+       -X \
```
Comment (kevincianfarini) [inline flyway/migrate.sh:19]: remove debug info.

**Review by jacknugent — APPROVED** [review]

**Comment by mcchang** [comment]
Nice!!

**Pull request merged**